File: TakeThePill/app/src/main/java/embeddedproject/takethepill/DrugEntity.java
package embeddedproject.takethepill;

import android.content.ContentValues;

import database.Str;

public class DrugEntity {

// VARIABILI

    private String nome;
    private String descrizione;
    private String tipo;
    private double prezzo;
    private int scorte;

//COSTRUTTORE
    public DrugEntity(String nome, String descrizione, String tipo, double prezzo, int scorte){

        this.nome=nome;
        this.descrizione=descrizione;
        this.tipo=tipo;
        this.prezzo=prezzo;
        this.scorte=scorte;
    }
    //costruttore di default

    public DrugEntity()
    {

        this.nome=null;
        this.descrizione=null;
        this.tipo=null;
        this.prezzo=0.0;
        this.scorte=0;
    }

//METODI GET e SET

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getDescrizione() {
        return descrizione;
    }

    public void setDescrizione(String descrizione) {
        this.descrizione = descrizione;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public double getPrezzo() {
        return prezzo;
    }

    public void setPrezzo(double prezzo) {
        this.prezzo = prezzo;
    }

    public int getScorte() {
        return scorte;
    }

    public void setScorte(int scorte) {
        this.scorte = scorte;
    }

    public ContentValues getAllValues()
    {
        ContentValues values =new ContentValues();

        values.put(Str.drugName,nome);
        values.put(Str.drugPrice,prezzo);
        values.put(Str.drugDescription,descrizione);
        values.put(Str.drugQuantities,scorte);
        values.put(Str.drugType,tipo);
        return values;
    }
}

File: TakeThePillKV/app/src/main/java/embeddedproject/takethepillkv/DrugEntity.kt
package embeddedproject.takethepillkv

import android.content.ContentValues;

import database.Str;

public class DrugEntity {

// VARIABILI
    private var nome:String?
    private var descrizione:String?
    private var tipo:String?
    private var prezzo:Double
    private var scorte:Int

//COSTRUTTORE
    constructor( nome :String, descrizione:String ,tipo:String ,prezzo: Double ,scorte: Int){

        this.nome=nome
        this.descrizione=descrizione
        this.tipo=tipo
        this.prezzo=prezzo
        this.scorte=scorte
    }


    constructor()
    {
        this.nome=null
        this.descrizione=null
        this.tipo=null
        this.prezzo=0.0
        this.scorte=0
    }

//METODI GET e SET
    fun getNome(): String? {
        return nome
    }

    fun setNome(nome: String) {
        this.nome = nome
    }

    fun getDescrizione(): String? {
        return descrizione
    }

    fun setDescrizione(descrizione: String) {
        this.descrizione = descrizione
    }

    fun getTipo(): String? {
        return tipo
    }

    fun setTipo(tipo: String?) {
        this.tipo = tipo
    }

    fun getPrezzo(): Double {
        return prezzo
    }

    fun setPrezzo(prezzo: Double) {
        this.prezzo = prezzo
    }

    fun getScorte(): Int {
        return scorte
    }

    fun setScorte(scorte: Int) {
        this.scorte = scorte
    }

   fun  getAllValues() : ContentValues
    {
        val values =ContentValues()
        val str=Str()
        values.put(str.drugName,nome)
        values.put(str.drugPrice,prezzo)
        values.put(str.drugDescription,descrizione)
        values.put(str.drugQuantities,scorte)
        values.put(str.drugType,tipo)
        return values
    }
}
